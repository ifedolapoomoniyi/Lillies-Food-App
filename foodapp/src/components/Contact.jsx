import React from "react";
import "../styles/Contact.css";
import { useState } from "react";

const Contact = () => {

    const [email, setEmail] = useState("");
    
    const handleEmailChange = (e) => {
        e.preventDefault();
        setEmail(e.target.value);
        console.log(email);
    };

    const options = {
		method: "POST",
		body: JSON.stringify({
            my_email: "ifedolapo0001@gmail.com",
            message: "Hello, welcome to Lillies, thank you for subscribing!",
            first_name: "Dear",
            last_name: "Customer",
            title: "Welcome to Lillies!",
            user_email: "ifedolapo0001@gmail.com"
        }),
	};

    const handleSubmit = (e) => {
		e.preventDefault();
		fetch("https://email-backend.cyclic.app/message/send", options)
			.then((response) => response.json())
			.then((response) => console.log(response))
			.catch((err) => console.error(err));
	};


	return (
		<div className="Contact">
			<div className="contact-text">
				<h3>Get notified when we update!</h3>
				<div>
					Get notified when we add new items to our specials menu,
					update our price list when we have promos!
				</div>
			</div>
			<div className="contact-input">
				<input
					type="email"
					className="contact-input-box"
					placeholder="randomuser@gmail.com"
                    onChange={handleEmailChange}
				/>
				<button className="contact-submit-btn" onClick={handleSubmit}>
					Get notified
				</button>
			</div>
		</div>
	);
};

export default Contact;
