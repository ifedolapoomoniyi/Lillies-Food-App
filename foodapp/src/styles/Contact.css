.Contact{
    display: flex;
    flex-direction: row;
    padding: 10px 0;
    color: white;
    justify-content: space-evenly;
    padding: 70px 0px;
}
.contact-text{
    max-width: 500px;
}

.contact-text h3{
    color: var(--secondary-color);
    font-size: 30px;
}
.contact-input-box{
    height: 33px;
    border-radius: 5px;
    min-width: 300px;
    font-size: 18px;
    padding: 5px 13px;

}
.contact-input{
    padding-top: 45px;

}
.contact-submit-btn{
    height: 50px;
    padding: 9px 15px;
    background-color: var(--secondary-color);
    border-radius: 5px;
    margin-left: 4px;
    font-size: 18px;
}
.contact-submit-btn:hover{
    opacity: 0.7;
}
button{
    cursor: pointer;
}

.contact-input input::placeholder{
    color: rgb(197, 197, 197);
}