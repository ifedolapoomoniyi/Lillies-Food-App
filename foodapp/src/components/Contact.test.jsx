import React from "react"; 
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import Contact from "./Contact";

test("renders component correctly", () => {
    const view = render(<Contact/>)
    expect(view).toMatchSnapshot();
})

test("renders text correctly", () => {
    const { getByText }= render(<Contact/>)
    // eslint-disable-next-line testing-library/prefer-screen-queries
    const text = getByText("Get notified");
    expect(text).toBeInTheDocument();
})