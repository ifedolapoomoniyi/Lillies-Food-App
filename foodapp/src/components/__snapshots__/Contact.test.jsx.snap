// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Contact"
      >
        <div
          class="contact-text"
        >
          <h3>
            Get notified when we update!
          </h3>
          <div>
            Get notified when we add new items to our specials menu, update our price list when we have promos!
          </div>
        </div>
        <div
          class="contact-input"
        >
          <input
            class="contact-input-box"
            placeholder="randomuser@gmail.com"
            type="email"
          />
          <button
            class="contact-submit-btn"
          >
            Get notified
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Contact"
    >
      <div
        class="contact-text"
      >
        <h3>
          Get notified when we update!
        </h3>
        <div>
          Get notified when we add new items to our specials menu, update our price list when we have promos!
        </div>
      </div>
      <div
        class="contact-input"
      >
        <input
          class="contact-input-box"
          placeholder="randomuser@gmail.com"
          type="email"
        />
        <button
          class="contact-submit-btn"
        >
          Get notified
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
